internal/store/store.go

```diff
@@ -0,0 +1,47 @@
+package store
+
+import (
+	"context"
+
+	"github.com/jackc/pgx/v5/pgxpool"
+)
+
+type Store struct {
+	*pgxpool.Pool
+}
+
+func NewStore(ctx context.Context, connString string) (Store, error) {
+	dbpool, err := pgxpool.New(ctx, connString)
+	if err != nil {
+		return Store{}, err
+	}
+	return Store{dbpool}, nil
+}
+
+func (db Store) CreateUsersTable(ctx context.Context) error {
+	_, err := db.Exec(ctx,
+		`CREATE TABLE IF NOT EXISTS users (
+			id bigint GENERATED ALWAYS AS IDENTITY,
+			login text NOT NULL UNIQUE,
+			password text NOT NULL)`)
+	return err
+}
+
+func (db Store) AddUser(login string, hash []byte) (int, error) {
+	row := db.QueryRow(context.Background(), "INSERT INTO users (login, password) VALUES ($1, $2) RETURNING id", login, hash)
+	var userID int
+	err := row.Scan(&userID)
+	if err != nil {
+		return 0, err
+	}
+	return userID, nil
+}
+
+func (db Store) GetUser(login string) (hash []byte, userID int, err error) {
```

Comment:
Мне кажется стоит ID возвращать первым параметром. Это будет консистентно с другим методом. Хотя почему-бы нам вообще не возвращать модель типа `User`?

```suggestion
func (db Store) GetUser(login string) (userID int, hash []byte, err error) {
```

Reply:
А как тогда быть с интерфейсом в handler.go? Его ведь не определить без импорта пакета с моделью?
```go
type Store interface {
	AddUser(login string, hash []byte) (int, error)
	GetUser(login string) ([]byte, int, error)
}
```